lib/auth/auth_with_roles_test.go

```diff
@@ -867,6 +867,56 @@ func TestAWSRolesAnywhereCredentialGenerationForApps(t *testing.T) {
 
 	require.Equal(t, roleARN, identity.AWSRoleARNs[0], "Expected AWS Role ARN to match the one requested")
 	require.JSONEq(t, `{"Version":1,"AccessKeyId":"aki","SecretAccessKey":"sak","SessionToken":"st","Expiration":"2025-06-25T12:07:02.474135Z"}`, identity.RouteToApp.AWSCredentialProcessCredentials)
+
+	t.Run("app using aws oidc integration does not fail", func(t *testing.T) {
```

Comment:
It's a little bit odd to me to have a test that is both a standalone test but also contains a single subtest.

Subtests are well suited towards table tests when you have some common setup and then a series of test scenarios.

This code almost looks like a completely independent test to me.